### PR TITLE
Add nydusd daemon command line builder

### PR DESCRIPTION
**Pull request by changweige**
Generate nydusd exec style command line via a daemonCommand object. It provides the flexibility to start nydusd as a library API. 
Move nydusd daemon startup logics to a new file `daemon_adapter.go`

### CODE REVIEW

File: pkg/daemon/command/command.go [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+	for _, o := range opts {
+		o(&cmd)
+	}
+
+	args := make([]string, 0, 32)
+	t := reflect.TypeOf(cmd)
+	v := reflect.ValueOf(cmd)
+
+	// Zero value will be skipped
+	if sc := v.FieldByName("Mode"); !sc.IsZero() {
```
Comment (imeoer) [inline pkg/daemon/command/command.go:48]: Can we handle `subcommand` instead of `Mode` here?

File: pkg/daemon/command/command.go [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+)
+
+type Opt = func(cmd *DaemonCommand)
+
+// Define how to build a command line to start a nydusd daemon
+type DaemonCommand struct {
+	// "singleton" "fuse"
+	Mode string `type:"subcommand"`
+	// "blobcache" "fscache" "virtiofs"
+	FscacheDriver  string `type:"param" name:"fscache" default:""`
```
Comment (imeoer) [inline pkg/daemon/command/command.go:22]: Seems we haven't handled the `default` tag, can we remove it first?

Comment (changweige) [reply pkg/daemon/command/command.go]: Yes, lets remove it for now

File: pkg/daemon/command/command.go [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+				continue
+			}
+		}
+	}
+
+	if subcommand != "" {
+		// Ensure subcommand is at the first place.
+		r := make([]string, 1)
+		r[0] = subcommand
+		args = append(r, args...)
```
Comment (imeoer) [inline pkg/daemon/command/command.go:91]: `args = append([]string{subcommand}, args...)`

File: pkg/daemon/command/command.go [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+
+			v := v.Field(i).Bool()
+
+			if v {
+				flag := fmt.Sprintf("--%s", tag.Get("name"))
+				args = append(args, flag)
+			} else {
+				continue
+			}
+		}
```
Comment (imeoer) [inline pkg/daemon/command/command.go:84]: Better to handle `default` and give an error message, in case of an invalid option field is added.

Comment (changweige) [reply pkg/daemon/command/command.go:84]: good point

Comment (changweige) [reply pkg/daemon/command/command.go:88]: fixed

File: pkg/daemon/command/command.go (hunk 3 above)
Comment (changweige) [reply pkg/daemon/command/command.go]: fixed

**Review by imeoer — APPROVED** [review]

**Pull request merged**